cmd/tools/testwick/testwicker.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	cmodel "github.com/mattermost/mattermost-cloud/model"
+	mmodel "github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// TestWicker data struct for test wicker scenarios
+type TestWicker struct {
+	provisionerClient *cmodel.Client
+	mmClient          *mmodel.Client4
+	logger            *logrus.Logger
+	installation      *cmodel.InstallationDTO
+	userID            string
+	channelID         string
+	teamID            string
+	Error             error
+}
+
+// NewTestWicker creates a testwicker
+func NewTestWicker(provisionerClient *cmodel.Client, logger *logrus.Logger) *TestWicker {
+	return &TestWicker{
+		logger:            logger,
+		provisionerClient: provisionerClient,
+	}
+}
+
+// CreateInstallation interacts with provisioner and uses the provided request
+// to create a new installation
+func (w *TestWicker) CreateInstallation(request *cmodel.CreateInstallationRequest) *TestWicker {
+	w.logger.WithField("DNS", request.DNS).Info("Creating installation")
+	if err := request.Validate(); err != nil {
+		w.Error = errors.Wrap(err, "CreateInstallationRequest.Validate")
+		return w
+	}
+	i, err := w.provisionerClient.CreateInstallation(request)
+	if err != nil {
+		w.Error = errors.Wrap(err, "client.CreateInstallation")
+		return w
+	}
+	w.installation = i
+	return w
+}
+
+// DeleteInstallation interacts with provisioner and uses the provided request
+// to delete an installation
+func (w *TestWicker) DeleteInstallation() *TestWicker {
+	w.logger.WithField("ID", w.installation.ID).Info("Deleting installation")
+	err := w.provisionerClient.DeleteInstallation(w.installation.ID)
+	if err != nil {
+		w.Error = errors.Wrap(err, "client.DeleteInstallation")
+		return w
+	}
+	return w
+}
+
+// PostMessage post messages to the different channels which are created
+// automatically by the wicker
+func (w *TestWicker) PostMessage(samples int) *TestWicker {
+	w.logger.WithField("DNS", w.installation.DNS).Info("Posting messages")
+	if w.channelID == "" {
+		w.Error = fmt.Errorf("failed to post message. You need to create channels first")
+		return w
+	}
+	for i := 0; i < samples; i++ {
+		_, response := w.mmClient.CreatePost(&mmodel.Post{
```

Comment:
Did not know there is a MM client like that 😄  +1 for using it.

Reply:
yeah which is amazing actually :)